ai/src/main/java/org/triplea/ai/doesnothing/DoesNothingAi.java

```diff
@@ -18,15 +19,10 @@
  * An AI implementation that takes no action except to purchase and place units according to very
  * simple rules.
  */
-public class DoesNothingAi extends AbstractBuiltInAi {
+public class DoesNothingAi extends AbstractAi {
 
-  public DoesNothingAi(final String name) {
-    super(name);
-  }
-
-  @Override
-  public PlayerTypes.Type getPlayerType() {
-    return PlayerTypes.DOES_NOTHING_AI;
+  public DoesNothingAi(final String name, final PlayerTypes.AiType playerType) {
+    super(name, playerType);
```

Comment:
In `AbstractAi`, isn't the player type always (and will always be): `  private final PlayerTypes.AiType playerType;`

In that case, is there a way we can avoid passing the `playerType` parameter if it will always have the same value?

Similarly, we should consider the 'name' value as well. I think the AI implementation ought to own the name and know what it's label would be. In this case it does not make sense to have a `name` parameter passed to it either. I suspect it'll take some work to remove the former parameter given how the game is setup, would you agree that is the right direction in general or is there a major obstacle or reason why the AI instance should have parameters?

Reply:
`playerType` is a newly constructed object that is generated in `HeadedGameRunner#gatherPlayerTypes`.  So each AI class will have its own `PlayerTypes.AiType` instance.  It isn't a hardcoded value.

`name` and `label` are two different things.  `name` is something like "Germany", "United States", "Italy", etc.  `label` is "Hard (AI)", "Fast (AI)", "Does Nothing (AI)".  The AbstractAi stores the `name` and the `PlayerTypes.Type` (and its child `PlayerTypes.AiType`) stores the label.  When you are selecting the users, the drop down is built from the `PlayerTypes.Type` list.  Once the game starts, the `PlayerTypes.Type` is used to create the actual player object (see `newPlayerWithName`).

So, these parameters (name and playerType) are required for the `AbstractAi`.

Reply:
Ah, in this case `name` is ambigious. The label is 'node name' and name is 'player name'. The latter is poor naming as it is historically called that simply because the single player game was created first, so a each faction or country was named a 'player' (which ceases to make sense once you are in a multiplayer where a 'player' is a person and their name is their call-handle). 

Regardless, 'name' should probably be clarified to be either `playerName` or `factionName`.

Reply:
As a newcomer, neither of those are any more clarifying than `name` :)  So what do you want it renamed to?

Reply:
`playerName` would work. I hate that refers to the faction (aka country), but it is what it is..